test/eth/next_interval_calculator.spec.js

```diff
@@ -1,189 +1,177 @@
+    assert.deepStrictEqual(toBlock, 100);
+  });
+
+  it('would not return full BLOCK_INTERVAL', () => {
+    const worker = new eth_worker.worker(constants);
+    worker.lastExportedBlock = 0;
+    worker.lastConfirmedBlock = 37;
+
+    const { fromBlock, toBlock } = nextIntervalCalculator(worker);
+    assert.deepStrictEqual(fromBlock, 1);
+    assert.deepStrictEqual(toBlock, 37);
+  });
+});
 
-        const resultSecond = await nextIntervalCalculator(worker);
-        assert.deepStrictEqual(resultSecond.success, true);
-        assert.deepStrictEqual(resultSecond.fromBlock, 2);
-        assert.deepStrictEqual(resultSecond.toBlock, 2);
-    });
+describe('various scenarios', () => {
+  it('fetched interval should not go backwards even if the blockchain node reports old block numbers', async () => {
+    const worker = new eth_worker.worker(constants);
+    worker.web3Wrapper = new MockWeb3Wrapper(100);
+    const context = await analyzeWorkerContext(worker);
+    assert.deepStrictEqual(context, WORK_SLEEP);
+    const result = nextIntervalCalculator(worker);
 
-    it('No sleep time if Node is ahead', async function () {
-        const worker = new eth_worker.worker(constants);
+    // Modify the last exported block as if we have consumed this interval
+    worker.lastExportedBlock = result.toBlock;
 
-        worker.lastExportedBlock = 1;
-        worker.lastConfirmedBlock = 2;
+    // Mock the Node to report an old number
+    worker.web3Wrapper.blockNumber = 90;
 
-        const resultSecond = await nextIntervalCalculator(worker);
-        assert.deepStrictEqual(resultSecond.success, true);
-        assert.deepStrictEqual(worker.sleepTimeMsec, 0);
-    });
+    const contextSecond = await analyzeWorkerContext(worker);
+    assert.deepStrictEqual(contextSecond, NO_WORK_SLEEP);
+  });
 
-    it('Node not called if Node is ahead', async function () {
-        const worker = new eth_worker.worker(constants);
-        worker.web3Wrapper = new MockIsCalledWeb3Wrapper();
+  it('fetched interval should not go backwards even if saved state is invalid', async () => {
+    const worker = new eth_worker.worker(constants);
+    worker.web3Wrapper = new MockWeb3Wrapper(4);
 
-        worker.lastExportedBlock = 1;
-        worker.lastConfirmedBlock = 2;
+    // Setup a situation where the exported block has exceeded the Node block
+    // Test is similar to the above but test that we already saved an old lastConfirmedBlock
+    worker.lastExportedBlock = 10;
+    worker.lastConfirmedBlock = 5;
 
-        await nextIntervalCalculator(worker);
-        assert.deepStrictEqual(worker.web3Wrapper.isCalled, false);
-    });
+    const context = await analyzeWorkerContext(worker);
+    assert.deepStrictEqual(context, NO_WORK_SLEEP);
+  });
 
-    it('Block interval is not exceeded', async function () {
-        const worker = new eth_worker.worker(constants);
+  it('interval is correct if the blockchain node is ahead', async () => {
```

Comment:
This test seems to cover functionality that has previously been tested separately in:
```
it('returns "work no sleep" case', async ()
```
and then
```
it('would not return full BLOCK_INTERVAL', ()
```

no need to have it if it adds nothing new